### PR TITLE
Make Workflow `dapr` Component internal

### PR DESCRIPTION
**Pull request by JoshVanL**
The internal workflow engine `dapr` component is currently being registered in the processor component store, making it visible to the user via the metadata registered component list and preventing the user creating a Component of the same name.

PR updates the workflow engine machinery to `Init` the component manually, bypassing the processor and preventing it being committed to the component store.

Adds integration test to ensure a `dapr` named component can be created by the user.

Updates tests that previously expected the workflow engine component to be registered in the component store.

closes https://github.com/dapr/dapr/issues/7411

/cc @mukundansundar @ASHIQUEMD

### CODE REVIEW

File: pkg/runtime/runtime.go [hunk 1]
```diff
@@ -644,12 +645,22 @@ func (a *DaprRuntime) initWorkflowEngine(ctx context.Context) error {
...
+		return fmt.Errorf("failed to initialize Dapr workflow component: %w", err)
 	}
-	return nil
+
+	log.Info("Workflow engine initialized.")
+
+	return a.runnerCloser.AddCloser(func() error {
+		return wfe.Close(wfengine.ComponentDefinition())
+	})
 }
```
Comment (JoshVanL) [inline pkg/runtime/runtime.go:664]: This is the main crux of the PR.

File: pkg/runtime/processor/processor.go [hunk 2]
```diff
@@ -209,12 +208,7 @@ func New(opts Options) *Processor {
 			components.CategorySecretStore:     secret,
 			components.CategoryStateStore:      state,
 			components.CategoryWorkflowBackend: wfbe,
-			components.CategoryWorkflow: workflow.New(workflow.Options{
-				Registry:       opts.Registry.Workflows(),
-				ComponentStore: opts.ComponentStore,
-				Meta:           opts.Meta,
-			}),
-			components.CategoryMiddleware: middleware.New(),
+			components.CategoryMiddleware:      middleware.New(),
```
Comment (JoshVanL) [inline pkg/runtime/processor/processor.go:211]: Ensures the workflow Component type cannot be added by the user.

**Review by ASHIQUEMD — APPROVED** [review]
LGTM

**Review by artursouza — APPROVED** [review]

**Pull request merged**